### PR TITLE
Synchronization update

### PR DESCRIPTION
**Pull request by chelseajohn**

### CODE REVIEW

File: DLplatform/worker.py [hunk 1]
```diff
@@ -236,6 +236,7 @@ def onCommunicatorMessageReceived(self,  routing_key, exchange, body):
             param = message['param']
             flags = message['flags']
             self._learner.setModel(param, flags)
+            self._communicator.learningLogger.logIntermediateModel(param) #logs the intermediate model
```
Comment (link-er) [inline DLplatform/worker.py]: I guess now you have it in two places, where you log the model?

File: DLplatform/learningLogger.py [hunk 2]
```diff
@@ -152,6 +165,21 @@ def logModel(self, filename : str, params: Parameters):
...
+
+    def logIntermediateModel(self,param: Parameters):
+        '''
+        Logs intermediate model, i.e., saves the parameters of intermediate model
+
+        Parameters
+        ----------
+        params - weights of an averaged model
+        '''
+        filename = 'IntermediateState_' +str(time.time())
```
Comment (link-er) [inline DLplatform/learningLogger.py]: Can you please add the if condition, that this model is saved only when you are in DEBUG mode (it is done also in the model saving that was there before)

File: DLplatform/worker.py (hunk 1 above)
Comment (chelseajohn) [reply DLplatform/worker.py]: You can see  in the new pull request that 'after' models are logged in DLplatform/worker.py : onCommunicationMessageReceived() function and 'before' models are logged in the setModel() function in DLplatform/learning/learner.py

File: DLplatform/learningLogger.py (hunk 2 above)
Comment (chelseajohn) [reply DLplatform/learningLogger.py]: I have added the If condition to log intermediate models only in DEBUG mode

File: DLplatform/worker.py (hunk 1 above)
Comment (link-er) [reply DLplatform/worker.py]: okey, I see

File: DLplatform/learningLogger.py (hunk 2 above)
Comment (link-er) [reply DLplatform/learningLogger.py]: thank you!

**Pull request merged**